src/Components/EnvironmentPicker/EnvironmentPicker.tsx

```diff
@@ -122,454 +115,462 @@ const EnvironmentPicker = (props: EnvironmentPickerProps) => {
-                                }
-                            }}
-                        />
-                    )}
-            </div>
-        );
-    };
+        return () => clearTimeout(dialogResettingValuesTimeoutRef.current);
+    }, []);
 
     const handleOnEditClick = useCallback(() => {
         toggleIsDialogHidden();
     }, []);
 
-    const handleOnEnvironmentUrlChange = useCallback(
-        (option, value) => {
-            if (option) {
-                setEnvironmentToEdit(option.data ?? option.text);
-            } else {
-                let newVal = value;
-                if (
-                    // let user enter hostname and gracefully append https protocol
-                    !newVal.startsWith('https://') &&
-                    ValidAdtHostSuffixes.some(
-                        (suffix) =>
-                            newVal.endsWith(suffix) ||
-                            newVal.endsWith(suffix + '/')
-                    )
-                ) {
-                    newVal = 'https://' + newVal;
-                }
-                setEnvironmentToEdit(newVal);
-                if (
-                    isValidUrlStr(newVal, 'environment') &&
-                    environments.findIndex((e: string | IADTInstance) =>
-                        typeof e === 'string'
-                            ? new URL(e).hostname === new URL(newVal).hostname
-                            : 'https://' + e.hostName ===
-                              new URL(newVal).hostname
-                    ) === -1
-                ) {
-                    setEnvironments(environments.concat(newVal));
-                }
-            }
-        },
-        [environments, isValidUrlStr]
-    );
-
-    const handleOnContainerUrlChange = useCallback(
-        (option, value) => {
-            let newVal = option ? option.text : value;
-            if (!newVal.startsWith('https://')) {
-                // let user enter hostname and gracefully append https protocol
-                try {
-                    const urlObj = new URL('https://' + newVal);
-                    if (
-                        ValidContainerHostSuffixes.some((suffix) =>
-                            urlObj.hostname.endsWith(suffix)
-                        ) &&
-                        urlObj.pathname !== '/'
-                    ) {
-                        newVal = 'https://' + newVal;
-                    }
-                } catch (error) {
-                    console.error('Not a valid URL string!');
-                }
-            }
-            setContainerUrlToEdit(newVal);
-            if (
-                isValidUrlStr(newVal, 'container') &&
-                containers.findIndex((e) => e === newVal) === -1
-            ) {
-                setContainers(containers.concat(newVal));
-            }
-        },
-        [containers, isValidUrlStr]
-    );
-
     const handleOnSave = useCallback(() => {
         if (props.onDismiss) {
             props.onDismiss();
```

Comment:
we should probably wait to notify the parent until we are done right? They could potentially unmount the EnvironmentPicker before it's done doing it's work.

Reply:
was already fixed: https://github.com/microsoft/iot-cardboard-js/blob/8cf9dcb84ce2ec3f7a62c82c265d7d37c8630aa9/src/Components/EnvironmentPicker/EnvironmentPicker.tsx#L413